test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -387,12 +388,32 @@ private void performFailoverTest() {
                                                         final AtomicBoolean stopWriteFlag, final AtomicLong eventWriteCount) {
         return CompletableFuture.runAsync(() -> {
             while (!stopWriteFlag.get()) {
-                Transaction<Long> transaction = null;
+                Transaction<Long> txnDebugReference = null;
+
                 try {
-                    transaction = retry
+                    Transaction<Long> transaction = retry
                             .retryingOn(MultiReaderTxnWriterWithFailoverTest.TxnCreationFailedException.class)
                             .throwingOn(RuntimeException.class)
                             .run(() -> createTransaction(writer, stopWriteFlag));
+                    txnDebugReference = transaction;
+                    AtomicBoolean txnIsDone = new AtomicBoolean(false);
+
+                    // Sets a recurrent delayed task to ping the txn. It exits when the
+                    // txn completes and no longer needs pinging
+                    FutureHelpers.loop(txnIsDone::get, () -> {
+                        return FutureHelpers.delayedTask(() -> {
+                            if (transaction.checkStatus() == Transaction.Status.OPEN) {
```

Comment:
no, the compiler doesn't like it that way, that's why I had to create another reference.